### PR TITLE
Bug fix: MultiOutSizeLinear masks outputs to 0 when dimension of  parameter >1 and dim * patch_size is not present in out_feat_size.

### LINKED ISSUE #44: weights_logits:4 the MultiOutSizeLinear.forward   out is alwayls zero
MultiOutSizeLinear.forward 
self.out_features_ls is [32, 64, 128, 256, 512]  
because  weights_logits:4  * [8, 16, 32, 64, 256]

when out_feat_size is 8,  
the "torch.eq(out_feat_size, feat_size).unsqueeze(-1) "  is always is False
then the out is alwayls zero.  is it right?

    def forward(
        self,
        x: Float[torch.Tensor, "*batch in_feat"],
        out_feat_size: Int[torch.Tensor, "*batch"],
    ) -> Float[torch.Tensor, "*batch max_feat"]:
        out = 0
        for idx, feat_size in enumerate(self.out_features_ls):
            weight = self.weight[idx] * self.mask[idx]
            bias = self.bias[idx] if self.bias is not None else 0
            out = out + (
                torch.eq(out_feat_size, feat_size).unsqueeze(-1)
                * (einsum(weight, x, "out inp, ... inp -> ... out") + bias)
            )
        return out

### PR DESCRIPTION
**Pull request by gorold**
This PR fixes a major bug for the `MultiOutSizeLinear` class.

When `dim > 1` in the following code snippet,
https://github.com/SalesforceAIResearch/uni2ts/blob/c4d75a3053c14e27ac4d875d7b31987abc1090a1/src/uni2ts/distribution/_base.py#L74-L85
it causes `for feat_size in self.out_features_ls` be `dim * patch_size`, whereas `out_feat_size` contains the patch sizes. This means that certain patch sizes could be ignored.
https://github.com/SalesforceAIResearch/uni2ts/blob/c4d75a3053c14e27ac4d875d7b31987abc1090a1/src/uni2ts/module/ts_embed.py#L172-L178

For the current configuration, this means that outputs with patch sizes 8 and 16 are always masked to 0.

Closes #44

### CODE REVIEW

**Comment by gorold** [comment]
lgtm :)
